File: apk_analyzer/apk_analyzer/dto/analysis_status.py
from dataclasses import dataclass


@dataclass
class HookStatus:
    type: str
    name: str
    code: int
    description: str


@dataclass
class PermissionStatus:
    name: str
    code: int
    description: str


@dataclass
class ManifestStatus:
    pkn: str
    permissions: list[PermissionStatus]


@dataclass
class AnalysisStatus:
    md5: str
    found_emulator: bool
    installed: bool
    started: bool
    finished: bool
    manifest: ManifestStatus
    analysis: list[HookStatus]

File: apkanalyzer-android/app/src/main/java/com/apk/analyzer/scanner/AnalysisStatus.java
package com.apk.analyzer.scanner;

import android.graphics.drawable.Drawable;

import java.util.List;

public class AnalysisStatus{
    public String md5;
    public String path;
    public boolean foundEmulator;
    public boolean installed;
    public boolean started;
    public boolean finished;
    public ManifestInfo manifest;
    public List<Hook> analysis;

    // Getters and setters
}

